### ISSUE TITLE
`/api/core/items` endpoint only returns "archived" Items when it should return all Items

### ISSUE THREAD

**Issue by tdonohue** [DSpace/DSpace#3325]
**Describe the bug**
Currently (as of 7.0), the `/api/core/items` endpoint only returns Items which are "in archive" and not withdrawn.  This means it excludes Workspace/Workflow items and withdrawn items.  This behavior is odd, because those item types **are available** via `/api/core/items/<:uuid>`, but they are currently only discoverable/findable via other related endpoints.

As this endpoint is Admin only & paginated, ideally the `/api/core/items` should return every Item in the system (i.e. everything in the `item` database table which is not deleted).

This bug does NOT impact the Angular UI, as it doesn't use this endpoint for any purposes at this time.  However, this may impact other clients which wish to use this endpoint to obtain access to all Items.

**Related work**
* Rediscovered during discussion of #3321 (In this PR, we temporarily "fixed" behavior to only return archived Items. Even though we feel that behavior is incorrect, it was easier to achieve quickly.
* Older, related discussions in PR #2406

**Comment by abollini** [comment]
it should took 2-4 hours including fixing/improve the IT to cover the right specification.
It would be appropriate to include at least one search method that will reproduce the behavior of the 7.0 so that admins can scroll all the archived items in the repository and/or the withdrawn items (so keep excluded workspace,workflow and template) without using the discovery configuration. This additional task and tests should take additional 4 hours

**Comment by tdonohue** [comment]
Thanks for the estimate @abollini .  I've added `4 hrs` to this ticket as the official estimate, as I think your second point may require more discussion.  I'm not yet sure if we need a new "out of the box" discovery configuration for only archived items, or if we simply should document that they should use the [existing `administrativeView` configuration](https://github.com/DSpace/DSpace/blob/main/dspace/config/spring/api/discovery.xml#L402) along with a filter to hide withdrawn items (as needed).  In any case, we can discuss that further once we move towards implementing this ticket.

**Comment by saschaszott** [comment]
This issue was moved to the DS backlog a year ago - is this an indicator that this issue will no longer being considered for DS 9? Currently, we need a case distinction in our DS API client when we want to perform bulk updates on DS items (regardless of their in_archive status).

**Comment by tdonohue** [comment]
@saschaszott : The fact that this was moved to the "backlog" over a year ago simply means that *at that time* there was an unclear priority on this task (see discussions above).  Any task can be moved from the backlog to a release board if/when we prioritize it or find a use case for it.

At this time, no one has found a use case for changing this endpoint because the search endpoint can be used to see *everything* (regardless of `in_archive`) status by using the `administrativeView`: https://github.com/DSpace/DSpace/blob/main/dspace/config/spring/api/discovery.xml#L643

So, in my opinion, your use case may be solved by using that `administrativeView`.  If it is not, then we'd need a clearer understanding of your use case....and we'd need to locate a volunteer for this ticket, obviously.

**Comment by saschaszott** [comment]
Thanks for your feedback, @tdonohue . Our current use case is that we want to modify items (add / remove / update metadata fields) regardless of the item's `in_archive` status. This is currently not possible via `/api/core/items` as this endpoint **does not consider** items which are in workflow or workspace.

In consequence, it means that I have to perform a case distinction in my program to decide which API endpoint is suitable for the PATCH / POST / PUT request (either `/api/core/items` or `/submission/workspaceitems` or `/workflow/workflowitems`).

This limitation is still mentioned in the description of the `/api/core/items` endpoint https://github.com/4Science/Rest7Contract/blob/main-cris/items.md

> In the future, this endpoint is likely to be updated to include all items, regardless of status, see https://github.com/DSpace/DSpace/issues/3325

To sum up, I would like to have an unqiue API endpoint that allows to modify items regardless of there current status.

**Comment by tdonohue** [comment]
@saschaszott : I believe you can already PATCH/POST/PUT **all items** via the `/api/core/items` endpoints.  For instance, our docs for PATCH have examples for how to reinstate a withdrawn item (where `is_archive` is false).  https://github.com/DSpace/RestContract/blob/main/items.md#replace

I believe this ticket is simply about whether to have the **root endpoint** `/api/core/items` list *every item in the system* (regardless of `is_archive` status), or whether it should only list items which have `is_archive: true` (current behavior).

So, I still believe your use case is already met in DSpace.  But, please let me know if I'm misunderstanding.

Regardless, this ticket **can be claimed** by someone who wants to volunteer to move it forward.  I'm not against changing this endpoint behavior, but it'd have to be done in a major release (as it may have unexpected side effects, if there's code that assumes `/api/core/items` will only return archived Items).
